framework/MCMC/MCMC.py

```diff
@@ -0,0 +1,293 @@
+    for key, value in self._updateValues.items():
+      totDim = self.variables2distributionsMapping[key]['totDim']
+      dist = self.distDict[key]
+      if totDim != 1:
+        self.raiseAnError(IOError,"Total dimension for given distribution {} should be 1".format(dist.type))
+      if value is None:
+        value = dist.rvs()
+        self._updateValues[key] = value
+
+  def localGenerateInput(self, model, myInput):
+    """
+      Provides the next sample to take.
+      After this method is called, the self.inputInfo should be ready to be sent
+      to the model
+      @ In, model, model instance, an instance of a model
+      @ In, myInput, list, a list of the original needed inputs for the model (e.g. list of files, etc.)
+      @ Out, None
+    """
+    for key, value in self._updateValues.items():
+      self.values[key] = value
+      self.inputInfo['SampledVarsPb'][key] = self.distDict[key].pdf(value)
+      self.inputInfo['ProbabilityWeight-' + key] = 1.
+    self.inputInfo['PointProbability'] = 1.0
+    self.inputInfo['ProbabilityWeight' ] = 1.0
+
+  def localFinalizeActualSampling(self, jobObject, model, myInput):
+    """
+      General function (available to all samplers) that finalize the sampling
+      calculation just ended. In this case, The function is aimed to check if
+      all the batch calculations have been performed
+      @ In, jobObject, instance, an instance of a JobHandler
+      @ In, model, model instance, it is the instance of a RAVEN model
+      @ In, myInput, list, the generating input
+      @ Out, None
+    """
+    AdaptiveSampler.localFinalizeActualSampling(self, jobObject, model, myInput)
+
+  def localStillReady(self, ready):
+    """
+      Determines if sampler is prepared to provide another input.  If not, and
+      if jobHandler is finished, this will end sampling.
+      @ In,  ready, bool, a boolean representing whether the caller is prepared for another input.
+      @ Out, ready, bool, a boolean representing whether the caller is prepared for another input.
+    """
+    ready = AdaptiveSampler.localStillReady(self, ready)
+    return ready
+
+  def _localHandleFailedRuns(self, failedRuns):
+    """
+      Specialized method for samplers to handle failed runs.  Defaults to failing runs.
+      @ In, failedRuns, list, list of JobHandler.ExternalRunner objects
+      @ Out, None
+    """
+    if len(failedRuns)>0:
+      self.raiseADebug('  Continuing with reduced-size MCMC sampling.')
+
+  def _formatSolutionExportVariableNames(self, acceptable):
+    """
+      Does magic formatting for variables, based on this class's needs.
+      Extend in inheritors as needed.
+      @ In, acceptable, set, set of acceptable entries for solution export for this entity
+      @ Out, acceptable, set, modified set of acceptable variables with all formatting complete
+    """
+    acceptable = AdaptiveSampler._formatSolutionExportVariableNames(self, acceptable)
+    return acceptable
```

Comment:
this method is not needed if you just call the baseclass method...you can just remove it